### PR TITLE
v1.1: Auto-update battery level every 5000 clocks, added color fill

### PR DESCRIPTION
**Pull request by jaredpellegrini**

### CODE REVIEW

File: savageRoller.ino [hunk 1]
```diff
@@ -264,8 +267,9 @@ void rollerHandleKeys() {
     }
     if (isNewlyPressed('b')) {
       // Show the battery screen.
-      firstRun = 1;
       currentPage = Page::Battery;
+      last_battery_check = 0;
+      batteryHandleDisplay();
```
Comment (kzaragoza) [inline savageRoller.ino:272]: Why move this here? Why not just do the state change and let the main loop call the right function?

**Review by LispEngineer — APPROVED** [review]
LGTM, Approved
(Not in depth review)

File: savageRoller.ino [hunk 2]
```diff
@@ -668,46 +672,59 @@ void splashHandleKeys() {
...
     }
   }
 }
 
 /* Display the currently estimated battery percentage on the screen. */
 void batteryHandleDisplay() {
-  if (!stateChange) {
+  // Only update the display if it's been more then 5000 clocks since last check
+  std::clock_t now = std::clock();
+  if (now - last_battery_check < 5000) {
```
Comment (kzaragoza) [inline savageRoller.ino:685]: The `now` value is in clock ticks per the [library function definition](https://en.cppreference.com/w/cpp/chrono/c/clock), and not necessarily milliseconds. You need to divide this by [CLOCKS_PER_SEC](https://en.cppreference.com/w/cpp/chrono/c/CLOCKS_PER_SEC) to get seconds.

File: savageRoller.ino [hunk 3]
```diff
@@ -668,46 +672,59 @@ void splashHandleKeys() {
...
+  // Start with a blank canvas, of course.
+  M5Cardputer.Display.clear();
+
+  // Show the text and percentage at the bottom of the screen.
+  M5Cardputer.Display.setTextColor(LIGHTGREY);
+  M5Cardputer.Display.setTextDatum(textdatum_t::bottom_center);
+  M5Cardputer.Display.drawString(level_label, displayWidth/2, displayHeight);
 
   // Draw the general outline of the battery.
   M5Cardputer.Display.drawRoundRect(left, top, width, height, radius);
```
Comment (kzaragoza) [inline savageRoller.ino:711]: You lost the call to `M5Cardputer.Display.setColor(LIGHTGREY);` Without that, you will get whatever color was last used to draw anything. Reinsert that line to make this color selection deterministic.

**Review by kzaragoza — CHANGES_REQUESTED** [review]

File: savageRoller.ino (hunk 2 above)
Comment (jaredpellegrini) [reply savageRoller.ino:685]: Yep, my comment here references 5000 clocks as opposed to seconds. I forgot to go back and change the other places, but I can update to calculate what 5 seconds would be and use that instead.

File: savageRoller.ino (hunk 3 above)
Comment (jaredpellegrini) [reply savageRoller.ino:711]: Good catch, I saw `setColor(LIGHTGREY)` and `setTextColor(LIGHTGREY)` and worked too quickly, assuming sameness. Will fix.

File: savageRoller.ino (hunk 1 above)
Comment (jaredpellegrini) [reply savageRoller.ino:272]: Testing and forgot to put it back? :)

**Pull request merged**